features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -852,14 +852,51 @@ internal class OtelTracerBuilderProviderTest {
         // Then
         argumentCaptor<(MutableMap<String, Any?>) -> Unit> {
             val traceContext: MutableMap<String, Any?> = mutableMapOf()
-            verify(mockSdkCore, times(1)).updateFeatureContext(eq(Feature.TRACING_FEATURE_NAME), capture())
+            verify(mockSdkCore, times(2)).updateFeatureContext(eq(Feature.TRACING_FEATURE_NAME), capture())
             lastValue.invoke(traceContext)
             assertThat(traceContext).doesNotContainKey(expectedActiveTraceContextName)
         }
     }
 
     // endregion
 
+    // region Tracing context
+
+    @Test
+    fun `M set opentelemetry as enabled in Context W build { TracingFeature enabled }`() {
+        // WHEN
+        testedOtelTracerProviderBuilder.build()
+
+        // THEN
+        argumentCaptor<(MutableMap<String, Any?>) -> Unit> {
+            val traceContext: MutableMap<String, Any?> = mutableMapOf()
+            verify(mockSdkCore, times(1)).updateFeatureContext(eq(Feature.TRACING_FEATURE_NAME), capture())
+            lastValue.invoke(traceContext)
+            assertThat(traceContext[TracingFeature.IS_OPENTELEMETRY_ENABLED_CONFIG_KEY]).isEqualTo(true)
+            assertThat(traceContext[TracingFeature.OPENTELEMETRY_API_VERSION_CONFIG_KEY])
+                .isEqualTo(BuildConfig.OPENTELEMETRY_API_VERSION_NAME)
+        }
+    }
+
+    @Test
+    fun `M set opentelemetry as enabled in Context W build { TracingFeature not enabled }`() {
+        // WHEN
+        testedOtelTracerProviderBuilder.build()
+        whenever(mockSdkCore.getFeature(Feature.TRACING_FEATURE_NAME)) doReturn null
+
+        // THEN
+        argumentCaptor<(MutableMap<String, Any?>) -> Unit> {
+            val traceContext: MutableMap<String, Any?> = mutableMapOf()
+            verify(mockSdkCore, times(1)).updateFeatureContext(eq(Feature.TRACING_FEATURE_NAME), capture())
+            lastValue.invoke(traceContext)
+            assertThat(traceContext[TracingFeature.IS_OPENTELEMETRY_ENABLED_CONFIG_KEY]).isEqualTo(true)
+            assertThat(traceContext[TracingFeature.OPENTELEMETRY_API_VERSION_CONFIG_KEY])
+                .isEqualTo(BuildConfig.OPENTELEMETRY_API_VERSION_NAME)
+        }
```

Comment:
the behavior shouldn't be like that, no context should be updated if feature is disabled, check the implementation https://github.com/DataDog/dd-sdk-android/blob/cff8b54d458f7cbf6f935deb7ff9b18aec23d9e3/dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/DatadogCore.kt#L193

Reply:
Yes I am aware of that but in that case all the other places where we actually call that method should be under `if(featureExists)` flag to be consistent. The only reason I am calling this method even if the feature is not registered is to be consistent with:
https://github.com/DataDog/dd-sdk-android/blob/cff8b54d458f7cbf6f935deb7ff9b18aec23d9e3/features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/AndroidTracer.kt#L57
and 
https://github.com/DataDog/dd-sdk-android/blob/cff8b54d458f7cbf6f935deb7ff9b18aec23d9e3/features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/AndroidTracer.kt#L68
and same 
thing for `OtelTracerProvider`.
The real question here is : in case the Trace feature is not registered should we return a workable AndroidTracer or OtelTracerProvider in the `build` method or just a NoOp ? Let's discuss this in the daily.

Reply:
I don't mean the need to check if feature registered or not before calling `updateFeatureContext`, I mean that this test name is wrong. The name is `M set opentelemetry as enabled in Context W build { TracingFeature not enabled }`, but no, if tracing feature is not registered, then nothing should be set, because the real implementation of `updateFeatureContext` won't do anything. We may assert only up to the point checking that `updateFeatureContext` was called, but not that callback to update values was executed (and test name should be updated).

But probably it makes no sense to even call `updateFeatureContext` if `tracingFeature = null`. By adding this check (and it makes sense, because `tracingFeature` is already in the method anyway) this test can be updated to the proper behavior.

Reply:
IMO it makes no sense to call it at all but in the same time I still argue that we should probably talk about returning NoOps in case trace feature is not initialized.